libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -1484,13 +1484,18 @@ bool AP_GPS::all_consistent(float &distance) const
 {
     // return true immediately if only one valid receiver
     if (num_instances <= 1 ||
-        drivers[0] == nullptr || params[0].type == GPS_TYPE_NONE) {
+        drivers[0] == nullptr || params[0].type == GPS_TYPE_NONE || GPS_MAX_INSTANCES <= 1) {
```

Comment:
Is this needed? We already check `num_instances`, can `num_instances` be larger than `GPS_MAX_INSTANCES`?

Reply:
The new code being a compiler-time check, the compiler could choose to remove the code block which will never be crossed.